src/etcd/client.py

```diff
@@ -174,6 +177,15 @@ def uri(protocol, host, port):
             _log.debug("Machines cache initialised to %s",
                        self._machines_cache)
 
+    def __del__(self):
+        """Clean up open connections"""
+        if self.http is not None:
```

Comment:
Just catch the case where self.http is not defined or not a urllib3.PoolManager instead?